src/evict/evict_inline.h

```diff
@@ -65,31 +72,40 @@ __wt_evict_read_gen_bump(WT_SESSION_IMPL *session, WT_PAGE *page)
 }
 
 /*
- * __wt_evict_read_gen_new --
+ * __wti_evict_read_gen_new --
  *     Get the read generation for a new page in memory.
  */
 static WT_INLINE void
-__wt_evict_read_gen_new(WT_SESSION_IMPL *session, WT_PAGE *page)
+__wti_evict_read_gen_new(WT_SESSION_IMPL *session, WT_PAGE *page)
 {
     __wt_atomic_store64(
       &page->read_gen, (__evict_read_gen(session) + S2C(session)->evict->read_gen_oldest) / 2);
 }
 
 /*
- * __wt_evict_cache_stuck --
- *     Indicate if the cache is stuck (i.e., eviction not making progress).
+ * __wti_evict_page_soon_flagged --
+ *     Return whether a read generation value makes a page eligible for immediate eviction. Read
+ *     generations reserve a range of low numbers for special meanings and currently - with the
+ *     exception of the generation not being set - these indicate the page may be evicted
+ *     immediately.
  */
 static WT_INLINE bool
-__wt_evict_cache_stuck(WT_SESSION_IMPL *session)
+__wti_evict_page_soon_flagged(uint64_t *readgen)
 {
-    WT_EVICT *evict;
-    uint32_t tmp_evict_aggressive_score;
+    uint64_t gen;
 
-    evict = S2C(session)->evict;
-    tmp_evict_aggressive_score = __wt_atomic_load32(&evict->evict_aggressive_score);
-    WT_ASSERT(session, tmp_evict_aggressive_score <= WT_EVICT_SCORE_MAX);
-    return (
-      tmp_evict_aggressive_score == WT_EVICT_SCORE_MAX && F_ISSET(evict, WT_EVICT_CACHE_HARD));
+    WT_READ_ONCE(gen, *readgen);
+    return (gen != WT_READGEN_NOTSET && gen < WT_READGEN_START_VALUE);
```

Comment:
I'll defer to you on the final decision. You have a lot more context on this change.

I'm wondering if we could define API functions
```
__wt_evict_readgen_is_init()
__wt_evict_readgen_is_evict_soon()
__wt_evict_readgen_is_wont_need()
```
or a generic
```
__wt_readgen_is(readgen, WT_READGEN_WONT_NEED)
```
here.

My reasoning is I don't see many new special `WT_READGEN_` values being added in future, and if they are perhaps we add a `WT_MUST_KEEP` value which would break the `< WT_READGEN_START_VALUE` check. 
This would let us be more specific in what we're checking. The line above for example would become
`return (__wt_evict_readgen_is(gen, WT_READGEN_OLDEST) || __wt_evict_readgen_is(gen, WT_READGEN_WONT_NEED));`

Reply:
I have made some changes in [0012113](https://github.com/wiredtiger/wiredtiger/pull/11087/commits/0012113bc68246830394367a1c264190c6bbdcb6) commit. Let me know what you think.

Since most individual checks for `READGEN_OLDEST` and `READGEN_WONT_NEED` are internal to eviction, with only one [instance](https://github.com/wiredtiger/wiredtiger/pull/11087/files#diff-9691a0692cb4c26132c4c27aa260cc6441c2ace4b04487d51769fe84869c624aR172) outside eviction checking just `READGEN_OLDEST`, I have made the folliwing changes - 
- Reverted [1fc5939](https://github.com/wiredtiger/wiredtiger/pull/11087/commits/1fc5939dc3d87cec047b84688ea46854f36cee3a) changes.
- Renamed `WT_READGEN_OLDEST` to `WT_READGEN_EVICT_SOON`.
- The setter for `WT_READGEN_EVICT_SOON` remains (`__wt_evict_page_soon`), and I’ve added a getter: `__wt_evict_page_is_soon`.
- Renamed `__wt_evict_soon_flagged` to `__wt_evict_is_soon_or_wont_need`.
- I didn’t change the return condition `return (gen != WT_READGEN_NOTSET && gen < WT_READGEN_START_VALUE)` to check only `WT_READGEN_EVICT_SOON` and `WONT_NEED`, as I’m unsure if  read_gen could be below 100.

Reply:
I will update the APIs in the description once this is approved.